### ISSUE TITLE
[BUG] Torrential rain causes majoor lag

### ISSUE THREAD

**Issue by Orionox** [LeafWulf/weatherfx#3]
**Module Version:** v1.2

**Before open any issue**

1) Enable the module setting _"Enable debugging"_ (I do not see this option)

**Describe the bug**
The torrential rain effects are causing pretty heavy lag.

**Foundry Version:** 10.286

**Game System:** 5e 2.0.3

**Additional context**
I have a pretty powerful computer and I don't get lag when setting my own weather effects, only when they are set by weatherFX

**Comment by paulo-roger** [comment]
Hi, do you mind to share what is the settings of your own weather effects for torrential rain?
I could test and update it.

**Comment by Orionox** [comment]
I'm not sure what you mean? Which settings do you want. Also, I do think it
had to do with the size of the map.

**Comment by paulo-roger** [comment]
I mean when you set your own weather effects
>Additional context
I have a pretty powerful computer and I don't get lag when setting my own weather effects, only when they are set by weatherFX

**Comment by Orionox** [comment]
Oh! I just don't get lag when using the FXMastwr module to set rain that
looks similar to yours. Give me like a day and I'll see if I can grab the
exact settings I use.

**Comment by paulo-roger** [comment]
Thanks!

**Comment by Orionox** [comment]
I think its linked to the size oof my maps. I was getting the lag on a map
4000x7000

**Comment by paulo-roger** [comment]
thanks for the feedback, I am trying to find whats the right size limit and add an warning for the user when they apply weather to a scene.

**Comment by Orionox** [comment]
After some additional testing it appeared to be grid size causing the lag.
When I went to grid size 100 instead of 50 all lag disappeared even at the
larger map sizes.

**Comment by paulo-roger** [comment]
WOW! I reproduced the same problem here! Amazing feedback, man! Thank you so much, in the v1.3.2 we shall have a warn for this.
Thank you!

**Comment by Auranim** [comment]
Same here - 15206x6969 (don't judge me)
grid size 68

(ps. adoro achar br do nada <3)

**Comment by Orionox** [comment]
Does the lag go away when you increase grid size?

**Comment by Auranim** [comment]
> 

Yep. I just doubled the grid size and it immediately went away. Then switched back to 68 and the fps tanked again. Switching it to 100 increases the fps but not as much as 134. 200 grid size essentially means 60fps.
Seems tied to grid size indeed and worsens proportionally to how small it is.
